src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -1363,6 +1403,171 @@ void MFEMSidreDataCollection::DeregisterField(const std::string& field_name)
+  const std::string& species_field_name,
+  const std::string& specset_name,
+  const std::string& matset_name,
+  const bool volume_dependent)
+{
+  auto iter = m_specset_associations.find(species_field_name);
+  if(iter != m_specset_associations.end())
+  {
+    SLIC_WARNING("Species field "
+                 << species_field_name
+                 << " has already been associated with a species set: "
+                 << iter->second);
+    return;
+  }
+  m_specset_associations[species_field_name] = specset_name;
+  Group* specset_grp = m_bp_grp->createGroup("specsets/" + specset_name);
+  specset_grp->createViewScalar("volume_dependent",
+                                static_cast<int8>(volume_dependent));
+  // Since we're creating the species set, associate it with a material set
+  specset_grp->createViewString("matset", matset_name);
+}
+
+void MFEMSidreDataCollection::AssociateMaterialDependentField(
+  const std::string& material_dependent_field_name,
+  const std::string& matset_name)
+{
+  auto iter = m_material_dependent_fields.find(material_dependent_field_name);
+  if(iter != m_material_dependent_fields.end())
+  {
+    SLIC_WARNING("Field " << material_dependent_field_name
+                          << " has already been labeled as material-dependent"
+                             " and associated with a material set: "
+                          << iter->second);
+    return;
+  }
+  m_material_dependent_fields[material_dependent_field_name] = matset_name;
+}
+
+View* MFEMSidreDataCollection::getFieldValuesView(const std::string& field_name)
+{
+  const std::string field_values_name = "fields/" + field_name + "/values";
+  View* values_view = nullptr;
+  if(m_bp_grp->hasView(field_values_name))
+  {
+    // Scalar-valued field
+    values_view = m_bp_grp->getView(field_values_name);
+  }
+  else if(m_bp_grp->hasGroup(field_values_name))
+  {
+    // Vector-valued field
+    values_view = m_bp_grp->getGroup(field_values_name)->getView("x0");
+  }
+  else
+  {
+    SLIC_WARNING("Field " << field_name << " was not registered");
+  }
+  return values_view;
+}
+
+void MFEMSidreDataCollection::checkForMaterialSet(const std::string& field_name)
+{
+  const auto tokens = detail::splitLastNTokens(field_name, 2, '_');
+  // Expecting [base_field_name, material_id]
+  if(tokens.size() != 2)
+  {
+    return;
+  }
+
+  auto iter = m_matset_associations.find(tokens[0]);
+  // If it hasn't been registered as a matset, it's not a volume fraction field
+  if(iter == m_matset_associations.end())
+  {
+    return;
+  }
+  const std::string matset_name = iter->second;
+
+  View* vol_fractions_view = getFieldValuesView(field_name);
+
+  Group* fractions_group =
+    alloc_group(m_bp_grp, "matsets/" + matset_name + "/volume_fractions");
+  View* matset_frac_view = fractions_group->copyView(vol_fractions_view);
+  matset_frac_view->rename(tokens[1]);
+  // FIXME: Do we need to add anything to the index group?
```

Comment:
>  // FIXME: Do we need to add anything to the index group?

Tag: @cyrush  -- I think this is a question for you

Reply:
Not quite sure, I assume this means the mesh bp index, which I believe should be generated by conduit?